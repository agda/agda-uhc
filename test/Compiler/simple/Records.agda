module Records where

open import Common.Nat
open import Common.IO
open import Common.Unit

record Test : Set where
  constructor mkTest
  field
    a b c : Nat

f : Test -> Nat
f (mkTest a b c) = a + b + c

open Test

g : Nat
g = (f (mkTest 34 12 54)) + (f (record {a = 100; b = 120; c = 140})) + a m + b m + c m
  where m = mkTest 200 300 400

main : IO Unit
main = printNat g
