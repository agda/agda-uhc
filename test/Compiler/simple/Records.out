EXECUTED_PROGRAM

ret > ExitSuccess
out > 1360
